### DISCUSSION TITLE
frontend 2 sockets at same time

### DISCUSSION THREAD

**Discussion by amchconsult** [socketio/socket.io#5023] [Q&A]
Hi, 
I am developing a web application.
One web sockets is connected to my node.js server no problem but also in front end I have add the ePOS_SDK_JavaScript_v2.27.0 file epos-2.27.0.js to connect to the thermal printer what also use socket.
So loading:
<script src="../assets/socket.io/dist/socket.io.min.js"></script>
<script src="../assets/ePOS_SDK_JavaScript_v2.27.0/epos-2.27.0.js"></script>

It said "io is not a function" coming from var io = io();
How do I do?
Thank you

**Comment by darrachequesne** [comment]
Hi! "io is not a function" usually means the script was not properly loaded. Could you please check?

**Reply by amchconsult** [reply]
Hi, thanks for answering. 
I have check and socket io is loading but:

```
<script src="../assets/socket.io/dist/socket.io.min.js"></script>
<script src="../assets/ePOS_SDK_JavaScript_v2.27.0/epos-2.27.0.js"></script>
```
1- If I load socket.io before epos from Epson, the socket.io is not working as per the source code of epos it is loading custom socket library to the io variable.

```
<script src="../assets/ePOS_SDK_JavaScript_v2.27.0/epos-2.27.0.js"></script>
<script src="../assets/socket.io/dist/socket.io.min.js"></script>
```
2- if I load epos from Epson first and then socket.io. Socket io works but the epos library for the thermal printer does not work.

I dont know how to avoid that to happen.
thank you
